code/FulpstationCode/body_camera_files/body_camera_procs.dm

```diff
@@ -0,0 +1,102 @@
+#define SEC_BODY_CAM_SOUND list('sound/machines/beep.ogg')
+#define SEC_BODY_CAM_REG_DELAY 0.5 SECONDS
+
+/obj/item/clothing/under/rank/security/Initialize()
+	. = ..()
+	builtInCamera = new (src)
+	builtInCamera.network = list("sec_bodycameras")
+	builtInCamera.internal_light = FALSE
+
+	RegisterSignal(src, COMSIG_ITEM_EQUIPPED, .proc/auto_register_bodycam)
+
+	addtimer(CALLBACK(src, /obj/item/clothing/under/rank/security.proc/auto_register_bodycam, null, SLOT_W_UNIFORM), SEC_BODY_CAM_REG_DELAY)
+
+/obj/item/clothing/under/rank/security/proc/auto_register_bodycam(mob/user, slot)
+	if(!builtInCamera)
+		return
+	if(slot != SLOT_W_UNIFORM)
+		return
+	if(!user)
+		if(ismob(loc))
+			user = loc
+		else
+			return
+	var/obj/item/card/id/I = user.get_idcard(TRUE)
+	if(!istype(I))
+		return
+	if(check_access(I))
+		var/id_name = I.registered_name
+		builtInCamera.c_tag = "-Body Camera: [id_name] ([I.assignment])"
+		camera_sound()
+		to_chat(user, "<span class='notice'>Security uniform body camera automatically registered to [id_name]</span>")
+
+/obj/item/clothing/under/rank/security/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(!builtInCamera)

```

Comment:
Correct me if I'm misreading, but wouldn't this be called even if you're clicking with something other than an ID? That doesn't seem intended, right?

Reply:
Built in camera should probably be shifted down until after the check for ID, but otherwise this seems fine.

Reply:
Then I think that's worth doing. Better than it triggering when you try to affix an armband, for example.